src/App.jsx

```diff
@@ -4,30 +4,37 @@ import { Header } from './Header';
 import { HomeArticles } from './HomeArticles';
 import { SingleArticle } from './SingleArticle';
 import { Routes, Route } from 'react-router-dom';
-import { ArticleComments } from './ArticleComments';
 
 function App() {
 	const [articles, setArticles] = useState(null);
+	const [comments, setComments] = useState([]);
+	const [isLoading, setIsLoading] = useState(true);
 	return (
 		<div>
 			<Header />
 			<Routes>
 				<Route
 					path='/'
 					element={
-						<HomeArticles articles={articles} setArticles={setArticles} />
+						<HomeArticles
+							articles={articles}
+							setArticles={setArticles}
+							isLoading={isLoading}
+							setIsLoading={setIsLoading}
+						/>
 					}
 				/>
 				<Route
 					path='/:article_id'
 					element={
-						<SingleArticle articles={articles} setArticles={setArticles} />
-					}
-				/>
-				<Route
-					path='/:article_id/comments'
-					element={
-						<ArticleComments articles={articles} setArticles={setArticles} />
+						<SingleArticle
```

Comment:
At this point, we're reaching prop drilling. Do you need to manage all this state in the top level? Or can you have different state management for each page? You're programming two pages here, why not have separate state for each one?